platform/darwin/src/MGLRasterSource.h

```diff
@@ -1,93 +1,84 @@
-#import "MGLSource.h"
-#import "MGLTypes.h"
+#import "MGLTileSource.h"
 
 #import <CoreGraphics/CoreGraphics.h>
 
-@class MGLTileSet;
-
 NS_ASSUME_NONNULL_BEGIN
 
 /**
+ An `NSNumber` object containing a floating-point number that specifies the
+ width and height (measured in points) of each tiled image in the raster source.
+ 
+ This number also corresponds to the width and height (measured in points) at
+ which the map displays each tile when the map’s zoom level is an integer. The
+ source scales its images up or down when the map’s zoom level falls between two
+ integers.
+ 
+ This option applies only to tiles hosted by Mapbox and accessed through version
+ 4 of the
+ <a href="https://www.mapbox.com/api-documentation/#maps">Mapbox Maps API</a>,
+ such as through `mapbox://<mapid>` URLs. This option has no effect on tiles
+ accessed through the
+ <a href="https://www.mapbox.com/api-documentation/#static">Mapbox Static API</a>
+ or hosted on third-party tile servers.
+ 
+ The only supported values for this option are 256 and 512. When the value is
+ 256, the source requests 1× images on a standard-resolution display and 2×
+ images on a Retina display, and each image is scaled to fit in a square
+ measuring 256 points on each side. When the value is 512, the source requests
+ 2× images regardless of the screen resolution, and each image is displayed in a
```

Comment:
The special 512 behavior for mapbox: URLs doesn’t fit into the coherent (and erroneous) model I originally thought it did. Explaining it will only make the option more confusing. I think we can omit this paragraph. mapbox: URLs will have a sensible default behavior, so it’s less likely that someone will run into this edge case.